cmake/clang-tidy.cmake

```diff
@@ -0,0 +1,13 @@
+if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+    find_program(CLANG_TIDY_EXECUTABLE NAMES clang-tidy-7 clang-tidy)
+    mark_as_advanced(CLANG_TIDY_EXECUTABLE)
+
+    if (${CLANG_TIDY_EXECUTABLE})
+        message(FATAL_ERROR "Clang-tidy not found")
+    else()
+        message(STATUS "Enabling clang-tidy")
+        set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_EXECUTABLE};-warnings-as-errors=*")
```

Comment:
Using warning-as-errors=* is a very strict policy, one could also reason to omit it (or do only specific checks).

Reply:
I would keep it enabled that way, since linting is for development on cpr. There we than can deal with those.